js/libs.js

```diff
@@ -3,33 +3,74 @@ Fliplet.Registry.set('fliplet-widget-notifications:1.0:core', function (data) {
   var DELAY = 30000;
 
   var appId = Fliplet.Env.get('appId');
-  var updateStorageKey = 'fl_notifications_' + appId;
-  var updateStorage;
+  var storageKey = 'fl_notifications_' + appId;
+  var storage;
   var instance;
+  var clearNewCountOnUpdate = false;
 
-  function saveUpdates(data) {
+  function saveCounts(data) {
     data = data || {};
 
-    updateStorage.updatedAt = data.updatedAt || new Date().getTime();
+    storage.updatedAt = Date.now();
+
+    if (clearNewCountOnUpdate || !storage.clearedAt) {
+      storage.clearedAt = Date.now();
+    }
+
     if (data.hasOwnProperty('unreadCount')) {
-      var unreadCount = 0;
-      unreadCount = parseInt(data.unreadCount, 0);
-      if (isNaN(unreadCount)) {
-        unreadCount = 0;
-      }
-      unreadCount = Math.max(0, unreadCount);
-      updateStorage.unreadCount = unreadCount;
+      storage.unreadCount = Math.max(0, parseInt(data.unreadCount, 10) || 0);
     }
 
     if (data.hasOwnProperty('newCount')) {
-      updateStorage.newCount = Math.min(updateStorage.unreadCount, data.newCount);
+      storage.newCount = Math.max(0, Math.min(storage.unreadCount, parseInt(data.newCount, 10) || 0));
     }
 
-    return Fliplet.App.Storage.set(updateStorageKey, updateStorage);
+    return Fliplet.App.Storage.set(storageKey, storage);
```

Comment:
you should save in the Storage, not the App Storage. You already namespaced the storageKey

Reply:
@squallstar Good point. Does this mean if I remove the `_{appId}` part in the key I can use `Fliplet.App.Storage`?

I suppose if I use `Fliplet.Storage` with the app ID in the storage key, it means the notifications will be accessible across different apps.

I'll probably go for the latter (as you suggested). Thoughts?

Reply:
Thanks. The storage is quite minimal at the moment and I can't imagine it expanding too much, so I'm going with App Storage with the key `flAppNotifications.